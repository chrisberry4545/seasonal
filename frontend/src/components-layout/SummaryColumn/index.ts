export * from './SummaryColumn';

export * from './SummaryColumn.interface';
