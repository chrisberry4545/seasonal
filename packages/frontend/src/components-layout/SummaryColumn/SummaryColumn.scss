@import './../../styles/settings';
@import './../../styles/tools';

.c-summary-column {
  width: 100%;
  margin-bottom: 2rem;
  @include mq($from: tablet) {
    width: 33%;
    padding: 0 0.75rem;
  }
}

  .c-summary-column__image {
    width: 100%;
    height: 150px;
    object-fit: cover;
    border-radius: $default-border-radius;
    margin-bottom: 1rem;
  }

  .c-summary-column__heading {
    margin-bottom: 0.75rem;
  }
