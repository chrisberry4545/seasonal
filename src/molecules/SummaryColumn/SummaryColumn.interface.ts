export interface ISummaryColumn {
  imageUrl?: string;
  text: string;
  title: string;
}
