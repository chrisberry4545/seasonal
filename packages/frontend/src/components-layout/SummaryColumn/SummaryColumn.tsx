import React, { FC } from 'react';

import './SummaryColumn.scss';

import {
  TextMedium,
  TextHeadingSmall
} from '@chrisb-dev/seasonal-shared-frontend-components';

import {
  ISummaryColumn
} from './SummaryColumn.interface';

export const SummaryColumn: FC<ISummaryColumn> = ({
  imageUrl,
  text,
  title
}) => (
  <div className='c-summary-column'>
    {
      imageUrl && <img src={imageUrl} className='c-summary-column__image' />
    }
    <h3 className='c-summary-column__heading'>
      <TextHeadingSmall>
        { title }
      </TextHeadingSmall>
    </h3>
    <TextMedium>
      { text }
    </TextMedium>
  </div>
);
