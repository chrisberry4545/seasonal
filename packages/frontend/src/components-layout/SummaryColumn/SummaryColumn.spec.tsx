import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { SummaryColumn } from '../SummaryColumn/SummaryColumn';

jest.mock('@chrisb-dev/seasonal-shared-frontend-components', () => ({
  TextHeadingSmall: () => 'TextHeadingSmall',
  TextMedium: () => 'TextMedium'
}));

describe('<SummaryColumn />', () => {
  let wrapper: ShallowWrapper;

  describe('when there is an image url', () => {
    beforeEach(() =>
      wrapper = shallow(
        <SummaryColumn
          title='Title'
          text='Some column text'
          imageUrl='https://image.com' />
      )
    );

    test('renders correctly', () => expect(wrapper).toMatchSnapshot());

  });

  describe('when there is no image url', () => {
    beforeEach(() =>
      wrapper = shallow(
        <SummaryColumn
          title='Title'
          text='Some column text' />
      )
    );

    test('do not display an image', () =>
      expect(wrapper.find('img').exists()).toBe(false));

  });

});
