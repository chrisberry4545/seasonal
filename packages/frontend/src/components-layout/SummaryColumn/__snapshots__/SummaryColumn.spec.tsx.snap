// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SummaryColumn /> when there is an image url renders correctly 1`] = `
<div
  className="c-summary-column"
>
  <img
    alt="Column item"
    className="c-summary-column__image"
    src="https://image.com"
  />
  <h3
    className="c-summary-column__heading"
  >
    <TextHeadingSmall>
      Title
    </TextHeadingSmall>
  </h3>
  <TextMedium>
    Some column text
  </TextMedium>
</div>
`;
